lib/components/narrative/metro/metro-itinerary.tsx

```diff
@@ -332,26 +332,17 @@ class MetroItinerary extends NarrativeItinerary {
     )
 
     const firstTransitStop = getFirstTransitLegStop(itinerary)
-
-    const renderAccessibilityHeader = (legs: Leg[]) => {
-      const routeLegs = legs.filter(removeInsignifigantWalkLegs)
-      const routeHeader = routeLegs.map((leg: Leg, index: number) => {
-        const legHeader = `${leg.routeShortName ? leg.routeShortName : ''} ${
-          leg.mode
-        }`
-        if (index === routeLegs.length) {
-          return legHeader
-        }
-        return ` ${legHeader}`
-      })
-      return routeHeader.toString()
-    }
-
-    const renderRouteBlocks = (legs: Leg[], firstOnly = false) => {
-      const routeBlocks = legs
-        .filter(removeInsignifigantWalkLegs)
+    const routeLegs = itinerary.legs.filter(removeInsignifigantWalkLegs)
+    const routeAndModeStrings = routeLegs.map((leg: Leg, index: number) => {
+      return `${
+        leg.routeShortName ? leg.routeShortName : ''
+      } ${getFormattedMode(leg.mode, intl)}`
+    })
```

Comment:
Much cleaner! Sadly though I think you'll have to use a new i18n key here -- so we can support RTL languages.

react-intl also supports most of the logic you're doing here! If you peek around you can see some examples of changing the string based on if variables are present

Reply:
Unfortunately the closest thing I found was the [Select](https://formatjs.io/docs/core-concepts/icu-syntax/#select-format) format, but I didn't have much luck filtering undefined variables. In the interest of time and i18n I went ahead and paired this header down to just the `FormattedList` of modes, although that may not be descriptive enough?